Apps/Database/src/Delegates/DBProfileDelegate.cs

```diff
@@ -279,5 +278,23 @@ public DBResult<IEnumerable<UserProfileHistory>> GetUserProfileHistories(string
             result.Status = DBStatusCode.Read;
             return result;
         }
+
+        /// <inheritdoc/>
+        public IDictionary<string, int> GetLoggedInUserYearOfBirthCounts(DateTime startDate, DateTime endDate)
+        {
+            Dictionary<string, int> yobCount = this.dbContext.UserProfile
+                .Select(x => new { x.HdId, x.LastLoginDateTime, x.YearOfBirth })
+                .Concat(
+                    this.dbContext.UserProfileHistory.Select(x => new { x.HdId, x.LastLoginDateTime, x.YearOfBirth }))
+                .Where(x => x.YearOfBirth != null && x.LastLoginDateTime >= startDate && x.LastLoginDateTime <= endDate)
```

Comment:
What is the generated SQL for this call?  Date is stored as UTC in the database but this is most likely being called from PST.  You all need to pass in an offset so the right time period is being queried.  See GetDailyLoggedInUsersCount but you will need to confirm when running the SQL itself.  To test this, put GetLoggedINUserYearOfBirthCounts call into DashboardService.GetDailyLoggedInUsersCount() and then put the following into your appsettings local json file to see the generated SQL in the console.     "Logging": {
        "LogLevel": {
            "Default": "Debug"
        },
        "SensitiveDataLogging": {
            "Enabled": true
        }
    },

Reply:
Added offset for UTC time.

Reply:
What is the generated SQL?

Right now both start date and end date param have been changed?  What about the UserProfile.LastLoginDateTime value that both start date and end date are comparing against? By adding an offset to both start date and end date, those parameters are now in the timezone you want them in but what about the UserProfile.LastLoginDateTime and UserProfileHistory.LastLoginDateTime?  Shouldn't last login date time also be adjusted to the time offset?  Example: UserProfile.LastLoginDateTime is October 10, 2022 06:59:00 UTC, which is 23:59:00 PST previous day (October 9, 2022).  If user enters October 5 and 10 at 5:00 am PST in the morning for start date and end date respectively, does this code change make start date October 4 and end date October 9 with the timespan offset change? If end date is changed to October 9. then it will  not pick up the record that had UserProfile.LastLoginDateTime of October 10, 2022 06:59:00 UTC.

Reply:
But last login datetime comes from the database so is already in UTC. So if I use the offset on the start date and end date that will come from the front end in pst, then the comparison is all in UTC.
Either that or apply the offset to the last login datetime only to make it pst like the start and end dates.  Should be the same result.

Reply:
Have a look at GetDailyLoggedInUsersCount()? That does an UTC adjustment.  GatewayDbContext.DateTrunc("days", x.LastLoginDateTime.AddMinutes(offset.TotalMinutes))

Reply:
I think you need both input parameters and what is being compared to in the database in the same timezone.

Reply:
The database datetimes are in UTC already are they not? Therefore LastLoginDateTime is in UTC when we select it from the database in the Select statements.
Look at DashboardService.GetRatingSummary, it is doing the same thing I am, it converts the startdate and enddate to UTC and then compares to the database datetime, which is already in UTC.
The example of GetDailyLoggedInUsersCount does not do this compare to start and end dates, it is returning the result in pst time for the front end to display because the LastLoginDateTime is in UTC in the database.

Reply:
What is the SQL saying?

Reply:
- Where clause is only being applied to UserProfileHistory.  It will need to be applied to **UserProfile** as well.
- For performance (analyze explain plan in PG Admin) you may need to modify the LINQ query to move the WHERE clause to associate with the SELECT DISTINCT 

- t."LastLoginDateTime" will be in UTC time.  Example: I just logged into Healthgateway at Wed Oct 12 11:03 PM.  In the database, the time is recorded as Thu Oct 13 06:03.  So if your start date is Monday, Oct 10 and end date is Wednesday, Oct 12, the above query should pick up a login that occurs on Wednesday, Oct 12 at 11:03 PM.

<img width="2503" alt="Screen Shot 2022-10-12 at 11 06 35 PM" src="https://user-images.githubusercontent.com/58790456/195515579-6c5929f8-2f06-4ba3-8558-6f4c2b9d1b60.png">

Reply:
I removed the OrderBy as per Mike's suggestion. This is the resulting code: 

![Screen Shot 2022-10-13 at 1 55 17 PM](https://user-images.githubusercontent.com/5408101/195708687-874bd921-79dc-4ee1-8d07-0ef9181f7d6f.png)

Here is the SQL that is created from it along with the Explain Analyze from PgAdmin:

![Screen Shot 2022-10-13 at 1 56 38 PM](https://user-images.githubusercontent.com/5408101/195708930-7245ee84-708c-446c-9970-c73e45703dc4.png)

1. I have tested it with local data and it is giving the correct results as far as I am expecting.
2. The Where clause is applied to both the UserProfile and UserProfileHistory tables.
3. The explain plan looks good from what I can see, no extra loops or scans. I could try to change it to do the Where clause on the Select Distinct if you think that is necessary.

Reply:
I think we can remove `lastLoginDate = GatewayDbContext.DateTrunc("days", x.LastLoginDateTime)` from the query.  We've already identified the users we are interested in by filtering on `LastLoginDateTime`.

Reply:
That is there because we don't want all the logins we want it to only count one per day if there are multiple.